Telegram/SourceFiles/media/media_audio_capture.cpp

```diff
@@ -569,14 +569,14 @@ void Instance::Inner::processFrame(qint32 offset, qint32 framesize) {
 	auto skipSamples = static_cast<int>(kCaptureSkipDuration * kCaptureFrequency / 1000);
 	auto fadeSamples = static_cast<int>(kCaptureFadeInDuration * kCaptureFrequency / 1000);
 	if (d->fullSamples < skipSamples + fadeSamples) {
-		qint32 fadedCnt = qMin(samplesCnt, skipSamples + fadeSamples - d->fullSamples);
+		qint32 fadedCnt = std::min(samplesCnt, skipSamples + fadeSamples - d->fullSamples);
 		double coef = 1. / fadeSamples, fadedFrom = d->fullSamples - skipSamples;
-		short *ptr = srcSamplesDataChannel, *zeroEnd = ptr + qMin(samplesCnt, qMax(0, skipSamples - d->fullSamples)), *end = ptr + fadedCnt;
+		short *ptr = srcSamplesDataChannel, *zeroEnd = ptr + std::min(samplesCnt, std::max(0, skipSamples - d->fullSamples)), *end = ptr + fadedCnt;
```

Comment:
clamp

Reply:
Clamp cant be used here. There may be situations it will lead to undefined behavior, this constuction doesnt guarantee that second argument of std:max greater 0 and first argument of std::min greater 0, or one of these greater than other

From specs of std::clamp
"The behavior is undefined if the value of lo is greater than hi"